src/download.rs

```diff
@@ -3,11 +3,14 @@ use client::GetResponse;
 use hyper::client::Client;
 use hyper::error::Error;
 use hyper::header::{ByteRangeSpec, Headers, Range};
-use pbr::{MultiBar, Pipe, ProgressBar};
+use pbr::{MultiBar, Pipe, ProgressBar, Units};
 use response::CheckResponseStatus;
 use std::cmp::min;
 use std::io::Read;
 use std::thread;
+use std::time::{Instant, Duration};
+
+const DOWNLOAD_BUFFER_BYTES: usize = 1024*64;
```

Comment:
A larger buffer means less context switches between the kernel and the snatch process.
The multiplication is just to it's clear that it is 64KiB.

2048byte buffer means 5120 loops per second on 10MiB/s
64KiB is 81 loops in the best case.

There are also Disk seek times on HDD and page sizes on SSDs to consider.